Bottom: 7d61e1806a7604fb0f6df839cd0485bc94caaf4a
Top:    7943e82f8cb4d00bac7cb811b0217e006be669c7
Author: Peter Zijlstra <peterz@infradead.org>
Date:   2018-05-30 16:22:45 +0200

sched/fair: Remove SIS_AVG_PROP

Hasn't been used in forever and doesn't give nice numbers anyway.

Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>


---

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index f5506623d1c1..393d02c043a7 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -6252,9 +6252,6 @@ static int select_idle_cpu(struct task_struct *p, struct sched_domain *sd, int t
 	}
 	avg_cost = this_sd->avg_scan_cost + 1;
 
-	if (sched_feat(SIS_AVG_CPU) && avg_idle < avg_cost)
-		return -1;
-
 	if (sched_feat(SIS_PROP)) {
 		u64 span_avg = sd->span_weight * avg_idle;
 		if (span_avg > sis_min_cores * avg_cost)
diff --git a/kernel/sched/features.h b/kernel/sched/features.h
index 8d2c7487b24c..3956c2316ad9 100644
--- a/kernel/sched/features.h
+++ b/kernel/sched/features.h
@@ -55,7 +55,6 @@ SCHED_FEAT(TTWU_QUEUE, true)
 /*
  * When doing wakeups, attempt to limit superfluous scans of the LLC domain.
  */
-SCHED_FEAT(SIS_AVG_CPU, false)
 SCHED_FEAT(SIS_PROP, true)
 
 SCHED_FEAT(SIS_AGE, true)
